src/main/java/cam72cam/immersiverailroading/entity/LocomotiveDiesel.java

```diff
@@ -86,9 +153,14 @@ public void onUpdate() {
 				}
 				
 				if (hasFuel) {
-					if (!idle.isPlaying()) {
+					if (!idle.isPlaying() && getTurnedOn() && Config.isFuelRequired(gauge)) {
```

Comment:
This can be simplified a lot

if (hasFuel && getTurnedOn())

That's all you need compared to the initial code.  The isFuelRequired is already in the hasFuel boolean